classes/statistics/PKPMetricsDAO.inc.php

```diff
@@ -354,7 +355,7 @@ protected function foreignKeyLookup($assocType, $assocId) {
 					$isFile = true;
 					$submissionId = $submissionFile->getData('submissionId');
 					if ($submissionFile->getData('assocType') == ASSOC_TYPE_REPRESENTATION) {
-						$representationId = $submissionFile->getData('assocId');
+						assert(!is_null($representationId));
```

Comment:
Should this throw an exception? What is the impact on the return array if no `$representationId` is set here?

Reply:
Hmmm... Yes, maybe it is best to throw an exception. It actually should never happen that the representationId is missing here. 
If it wouldn't be set the entry in the DB table would be null -- so I think nothing bad, but wrong entry in the DB...